tasks/eva_2121/aggregate_table.py

```diff
@@ -0,0 +1,226 @@
+#!/usr/bin/env python
+import csv
+import glob
+import logging
+import os
+import subprocess
+import sys
+from argparse import ArgumentParser
+from collections import defaultdict
+
+assigned_number_variant = 0
+eva_accession_path = ''
+
+logging.basicConfig(stream=sys.stdout, level=logging.INFO, format='%(asctime)-15s %(levelname)s %(message)s')
+logger = logging.getLogger(__name__)
+
+
+def run_command_with_output(command_description, command, return_process_output=False):
+    process_output = ""
+
+    logger.info("Starting process: " + command_description)
+    logger.info("Running command: " + command)
+
+    with subprocess.Popen(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE, bufsize=1, universal_newlines=True,
+                          shell=True) as process:
+        for line in iter(process.stdout.readline, ''):
+            line = str(line).rstrip()
+            logger.info(line)
+            if return_process_output:
+                process_output += line + "\n"
+        for line in iter(process.stderr.readline, ''):
+            line = str(line).rstrip()
+            logger.error(line)
+    if process.returncode != 0:
+        logger.error(command_description + " failed! Refer to the error messages for details.")
+        raise subprocess.CalledProcessError(process.returncode, process.args)
+    else:
+        logger.info(command_description + " - completed successfully")
+    if return_process_output:
+        return process_output
+
+
+def parse_input(input_file):
+    """
+    Parse the input with a dict reader and returns tthe data aggregated per taxid
+    """
+    data_per_taxid = defaultdict(list)
+    data_per_taxid_and_assembly = defaultdict(list)
+
+    with open(input_file) as open_file:
+        reader = csv.DictReader(open_file, delimiter='\t')
+        for row in reader:
+            if row['Decision to include'] == 'yes':
+                data_per_taxid_and_assembly[(row['Taxid'], row['Assembly'])].append(row)
+                data_per_taxid[row['Taxid']].append(row)
+    return data_per_taxid, data_per_taxid_and_assembly
+
+
+def assign_tempmongo_host_round_robin(number_variants, total_number_of_variant):
+    global assigned_number_variant
+    assigned_number_variant += number_variants
+    instance_number = int((assigned_number_variant // ((total_number_of_variant + 1) / 10)) + 1)
```

Comment:
I guess this ensures some equitable distribution of the load to different instances based on the number of variants?

Reply:
Yes this is the idea albeit not necessarily a clever one. If there is a more sensible way to split the data across, please let me know.

Reply:
While this automatic assignment of species to instances is fine for the most part, I think the species that share assemblies should be in the same instance (to facilitate clustering), correct?

Reply:
There should not be any species that share assemblies in this release. The release code would not support it.